Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionControllerCocoa.mm

```diff
@@ -44,10 +46,34 @@
     if (outError)
         *outError = nil;
 
-    if (!extensionContext.load(*this, outError))
+    if (!m_extensionContexts.add(extensionContext)) {
+        if (outError)
+            *outError = extensionContext.createError(WebExtensionContext::Error::AlreadyLoaded);
+        return false;
+    }
+
+    if (!m_extensionContextBaseURLMap.add(extensionContext.baseURL(), extensionContext)) {
+        m_extensionContexts.remove(extensionContext);
+        if (outError)
+            *outError = extensionContext.createError(WebExtensionContext::Error::BaseURLTaken);
+        return false;
+    }
+
+    if (!extensionContext.load(*this, outError)) {
```

Comment:
Do we want to do anything with `outError`? Or would any error be logged/vended elsewhere?

Reply:
It is just returned to the API call site, and is the primary way Swift handles errors. These turn into a `throws` `func` in Swift.